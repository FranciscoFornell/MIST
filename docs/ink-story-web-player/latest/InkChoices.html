<!DOCTYPE html>
<html lang="en">

<head>
  
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
<<<<<<< Updated upstream
  <title>Docs: Simple Ink Story Web Player InkChoices</title>
=======
  <title>Docs: Ink Story Web Player InkChoices</title>
>>>>>>> Stashed changes

  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>
  <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
  <script src="./build/entry.js"></script>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
  <![endif]-->
  <link href="https://fonts.googleapis.com/css?family=Roboto:100,400,700|Inconsolata,700" rel="stylesheet">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <link type="text/css" rel="stylesheet" href="https://jmblog.github.io/color-themes-for-google-code-prettify/themes/tomorrow-night.min.css">
  <link type="text/css" rel="stylesheet" href="styles/app.min.css">
  <link type="text/css" rel="stylesheet" href="styles/iframe.css">
  <link type="text/css" rel="stylesheet" href="">
  <script async defer src="https://buttons.github.io/buttons.js"></script>

  
</head>



<body class="layout small-header">
    <div id="stickyNavbarOverlay"></div>
    

<div class="top-nav">
    <div class="inner">
        <a id="hamburger" role="button" class="navbar-burger" aria-label="menu" aria-expanded="false">
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
        </a>
        <div class="logo">
            
             
                <a href="index.html">
<<<<<<< Updated upstream
                    <h1 class="navbar-item">Simple Ink Story Web Player</h1>
=======
                    <h1 class="navbar-item">Ink Story Web Player</h1>
>>>>>>> Stashed changes
                </a>
            
        </div>
        <div class="menu">
            
            <div class="navigation">
                <a
                    href="index.html"
                    class="link"
                >
                    API Documentation
                </a>
                
                 
                    
                        <a
                            class="link user-link "
                            href="https://github.com/FranciscoFornell/MIST/tree/master/packages/ink-story-web-player"
                        >
                            Github
                        </a>
                    
                
                
            </div>
        </div>
    </div>
</div>
    <div id="main">
        <div
            class="sidebar "
            id="sidebarNav"
        >
            
            <nav>
                
                    <h2><a href="index.html">Documentation</a></h2><div class="category"><h3>Components</h3><ul><li><a href="InkButton.html">InkButton</a></li><li><a href="InkChoices.html">InkChoices</a></li><li><a href="InkHeader.html">InkHeader</a></li><li><a href="InkLines.html">InkLines</a></li><li><a href="InkStoryWebPlayer.html">InkStoryWebPlayer</a></li></ul></div>
                
            </nav>
        </div>
        <div class="core" id="main-content-wrapper">
            <div class="content">
                <header class="page-title">
                    <p>Components</p>
                    <h1>InkChoices</h1>
                </header>
                




<section>

<header>
    
        <h2>
            &ltInkChoices <i>choices</i> <i>clickHandler</i> /&gt
        </h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        
            <div class='vertical-section'>
                <div class="members">
                    <div class="member">
                        <div class=name>
                            <span class="tag">Constructor</span>
                        </div>
                        


    
    <h4 class="name" id="InkChoices">
        <a class="href-link" href="#InkChoices">#</a>
        
        <span class="code-name">
            
                &ltInkChoices <i>choices</i> <i>clickHandler</i> /&gt
            
        </span>
    </h4>
    

    
    
    <div class="description">
        Simple Ink Web Player Choices
    </div>
    







    
        <h5>PropTypes:</h5>
        
<div class="table-container">
  <table class="params table">
    <thead>
    <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Required</th>
        <th>Description</th>
        <th class="last">Default</th>
    </tr>
    </thead>

    <tbody>
    
          <tr class="deep-level-0">
            <td class="name"><code>choices</code></td>
            <td>
              arrayOf
            </td>
            <td>Yes</td>
            <td>An array of ink choices</td>
            <td></td>
          </tr>
          
    
          <tr class="deep-level-0">
            <td class="name"><code>clickHandler</code></td>
            <td>
              func
            </td>
            <td>No</td>
            <td>The function to be called with the choice index as param when the choice button is clicked</td>
            <td></td>
          </tr>
          
    
    </tbody>
  </table>
</div>
    







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="InkChoices_index.js.html" class="button">View Source</a>
            <span>
                <a href="InkChoices_index.js.html">InkChoices/index.js</a>, <a href="InkChoices_index.js.html#line21">line 21</a>
            </span>
        </p>
    
</dl>





















    <h5>Example</h5>
    
    
        

<div id='component0'></div>

<script>
  ReactDOM.render(ReactWrapper({
    example: "const choices = [\n  { text: 'Choice 1', index: 1 },\n  { text: 'Choice 2', index: 2 },\n];\nreturn (\n  <InkChoices\n    choices={choices}\n    clickHandler={(index) => alert(`Choice ${index} selected!`)}\n  />\n);",
    uniqId: "component0",
  }), document.getElementById('component0'));
</script>
    



                    </div>
                </div>
            </div>
        
    
    </div>
    
        
    
    

    

    

    

    

    

    

    

    

    
</article>

</section>




            </div>
            
            <footer class="footer">
                <div class="content has-text-centered">
                    <p>Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.6.6</a></p>
                    <p class="sidebar-created-by">
                        <a href="https://github.com/SoftwareBrothers/better-docs" target="_blank">BetterDocs theme</a> provided with <i class="fas fa-heart"></i> by 
                        <a href="http://softwarebrothers.co" target="_blank">SoftwareBrothers - JavaScript Development Agency</a>
                    </p>
                </div>
            </footer>
            
        </div>
        <div id="side-nav" class="side-nav">
        </div>
    </div>
<script src="scripts/app.min.js"></script>
<script>PR.prettyPrint();</script>
<script src="scripts/linenumber.js"> </script>

</body>
</html>