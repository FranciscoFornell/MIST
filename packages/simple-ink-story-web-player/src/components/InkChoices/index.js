import React from 'react';
import PropTypes from 'prop-types';
import InkButton from '../InkButton';

/**
 * Simple Ink Web Player Choices
 * @component
 * @example
 * const choices = [
 *   { text: 'Choice 1', index: 1 },
 *   { text: 'Choice 2', index: 2 },
 * ];
 * return (
 *   <InkChoices
 *     choices={choices}
 *     clickHandler={(index) => alert(`Choice ${index} selected!`)}
 *   />
 * );
 */

const InkChoices = ({ choices, clickHandler }) => (
  <React.Fragment>
    {choices.map(({ text, index }) => (
      <InkButton key={index} onClick={() => clickHandler(index)} label={text} />
    ))}
  </React.Fragment>
);

InkChoices.propTypes = {
  /**
   * An array of ink choices
   */
  choices: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      index: PropTypes.number.isRequired,
    })
  ).isRequired,
  /**
   * The function to be called with the choice index as param when the choice button is clicked
   */
  clickHandler: PropTypes.func,
};

export default InkChoices;
