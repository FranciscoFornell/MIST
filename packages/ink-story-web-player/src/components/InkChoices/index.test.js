import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import InkChoices from './index';

const onClick = jest.fn();
let choices = [];

const renderComponent = () => {
  const { container, getByText } = render(
    <InkChoices choices={choices} clickHandler={onClick} />
  );
  return {
    snapshot: container.innerHTML,
    clickOn: (text) => {
      fireEvent.click(getByText(text));
    },
  };
};

afterEach(() => {
  jest.clearAllMocks();
  choices = [];
});
afterAll(() => {
  jest.restoreAllMocks();
});

describe('InkChoices component', () => {
  it('should render no buttons if no choices are given', () => {
    expect(renderComponent().snapshot).toMatchSnapshot();
  });
  it('should render as many buttons as choices are given', () => {
    choices = [
      { index: 1, text: 'text 1' },
      { index: 2, text: 'text 2' },
    ];
    expect(renderComponent().snapshot).toMatchSnapshot();
  });
  it('should call the given onClick callback, with the correct index when the button is clicked', () => {
    choices = [
      { index: 1, text: 'test-choice-1' },
      { index: 2, text: 'test-choice-2' },
    ];
    const { clickOn } = renderComponent();
    clickOn('test-choice-1');
    expect(onClick).toHaveBeenCalledWith(1);
    onClick.mockClear();
    clickOn('test-choice-2');
    expect(onClick).toHaveBeenCalledWith(2);
  });
});
