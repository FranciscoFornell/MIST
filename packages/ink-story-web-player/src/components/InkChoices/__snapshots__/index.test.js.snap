// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InkChoices component should render as many buttons as choices are given 1`] = `
<button>
  text 1
</button>
<button>
  text 2
</button>
`;

exports[`InkChoices component should render no buttons if no choices are given 1`] = `""`;
